lib/metababel/bt2_trace_class_generator.rb

```diff
@@ -717,6 +802,22 @@ def get_getter(variable:, arg_variables:)
         end
       end
     end
+
+    def match?(field_class)
+      field_class.members.all? do |rhs_member| 
+        members_matched = @members.find_all { |member| member.match?(rhs_member) }
+        raise "rhs_member '#{rhs_member}' must match one member '#{ members_matched.length }' matched." unless members_matched.length < 2
+        members_matched.length == 1
+      end
+    end
+
+    def get_args(field_class)
+      field_class.members.filter { |member| member.extract }.map do |rhs_member| 
+        members_matched = @members.find_all { |member| member.match?(rhs_member) }
+        raise "rhs_member '#{rhs_member}' must match one member '#{ members_matched.length }' matched." unless members_matched.length == 1
+        members_matched
+      end.flatten.map(&:get_arg)
+    end
```

Comment:
Will it be better if `match` return either False or the list of filter it matches?
s? I may avoid code duplication ( and add just one function not 2 :)

Or just use `get_args` who return empty list if not matching.
What does `extract` doe?

I found this function a little complicated with the filter + extract + flatten. A little comment will be appreciated to explain the goal of this function